### ISSUE TITLE
National correlation - how is it calculated?

### ISSUE THREAD

**Issue by andagit** [metno/pyaerocom#820]
For CAMS2-83 ie. EMEP SO2 R for GB is very low (-0.02), but no stations (4of5) have R lower than 0.1? (The 5th station has NaN - I think)

**Comment by lewisblake** [comment]
Leaving this link here for future reference: https://cams2-83.aeroval.met.no/evaluation.php?project=cams2-83&exp_name=forecast-last-seasons&station=United+Kingdom&par=concso2#

**Comment by lewisblake** [comment]
1) Document what the national correlations should be (i.e., is it the average of the correlations values at each station, or is it the correlation of all data points in the country (spatio-temporal correlation)?)
2) Test to see if what is begin computed is actually what it's supposed to be

**Comment by lewisblake** [comment]
I just spoke to @AugustinMortier about this. COntinuing our example from earlier of Greece with two stations. He says that for each timestep, the country-wide correlations are computed as the correlation between the averaged observed values and averaged model output at the station locations, and therefore we should not expect that the country-wide correlations, match the average correlations at the stations. He also says his most recent paper explains the methodology of this

**Comment by andagit** [comment]
Still it is a bit misleading or somewhat unclear what about this 5th station in GB that R is not calculated for(?) - is it a part of the country-wide correlation calculation or not - if you look at other stats - the 5th station has got 1 measurement(!) - should PyAerocom then take this station into account at all?

**Comment by lewisblake** [comment]
The minimum number of observations allowed is set in the config file, so we would need to see exactly what is specified there. With regards to the correlation though, the reason thi station with one obs appears in the statistics for other stations and not the correlation, is because to compute the correlation, you need to standardize by the sample standard deviation. The standard deviation of a single number is zero, and so the correlation and Spearman correlation can not be computed because they are not defined.

**Comment by charlienegri** [comment]
```
min_num_obs=dict(
         # yearly=dict(monthly=9),
         # monthly=dict(daily=21, weekly=3),
         daily=dict(hourly=18),
     ),
```
https://github.com/metno/pyaerocom/blob/temporary-cams2-83/pyaerocom/scripts/cams2_83/config.py#L102

**Comment by lewisblake** [comment]
Thanks, @charlienegri. So this one station is included because there is one day where there is more than 18 hours of observations, but our filters do not seem to account for the total number of days which make this criterion to compute daily statistics. It also appears in the computation of the monthly stats because that filter has been commented out, as we can see :)

**Comment by lewisblake** [comment]
@andagit If this explanation is sufficient please let us know so we can close the issue. Thanks

**Comment by andagit** [comment]
Yes I suppose so, and I guess @michaelgau is satisfied with this answer as well?
